engine/engine/src/engine.cpp

```diff
@@ -2017,14 +2017,37 @@ void dmEngineFinalize()
     dmSocket::Finalize();
 }
 
+const char* ParseArgOneOperand(const char* arg_str, int argc, char *argv[])
+{
+    for (int i = 0; i < argc; ++i)
+    {
+        const char* arg = argv[i];
+        if (strncmp(arg_str, arg, sizeof(arg_str)-1) == 0)
+        {
+            const char* res = strchr(arg, '=');
+            if (res)
+            {
+                return res + 1;
+            }
+        }
+    }
+
+    return 0;
+}
+
+
 dmEngine::HEngine dmEngineCreate(int argc, char *argv[])
 {
-    if (!dmGraphics::Initialize())
+    const char* arg_adapter_type = ParseArgOneOperand("--graphics-adapter", argc, argv);
+
+    if (!dmGraphics::Initialize(arg_adapter_type))
     {
-        dmLogError("Could not initialize graphics.");
+        dmLogError("Could not initialize graphics. No graphics adapter was found.");
         return 0;
     }
 
+    dmLogInfo("Initialised graphics device '%s'", dmGraphics::GetGraphicsAdapterTypeLiteral(dmGraphics::GetAdapterType()));
```

Comment:
I'm undecided on where this output should be.
On the one hand, I think it's good to "be in control" from the engine side, but on the other, I think this might just as well be a log output from within the library.

E.g. if we use the graphics library in another app (test_app_vulkan)

Also, in this case, adding the two functions GetGraphicsAdapterTypeLiteral()/GetAdapterType() to the graphics header could be delayed untill later since I'm not sure they are needed elsewhere?

Reply:
Yeah that's true, I just figured since we do something similar for the sound adapter, but it makes more sense to print it when the graphics has been initialized. I'll fix that